### ISSUE TITLE
[rv_dm,dv] Implement sec_cm testplan

### ISSUE THREAD

**Issue by rswarbrick** [lowRISC/opentitan#23721]
### Description

The testplan contains 8 testpoints and they are not yet implemented, which is a requirement for V2S status.

**Comment by johngt** [comment]
Based on the approval of the TC this effort is deferred to at least M6.
Making the Milestone change now.
@moidx / @vogelpi  / @andreaskurth

**Comment by rswarbrick** [comment]
This is now mostly implemented. Remaining testpoints to implement:
- `sec_cm_sba_tl_lc_gate_fsm_sparse`
- `sec_cm_mem_tl_lc_gate_fsm_sparse`
- `sec_cm_exec_ctrl_mubi`

The first two will be a case of injecting an error into a state register and checking things lock up (maybe half a day each). The third is checking that the multi-bit signal  we use for debug control handles invalid states correctly. This is actually implemented already (once #24158 is merged), so the only change needed is tweaking the testplan.

Total effort: 1 day.

**Comment by andreaskurth** [comment]
Moving to M7 as P1, as approved by TC today

**Comment by vogelpi** [comment]
I think this work is all done now and I've reviewed quite a few of these PRs and I did a full review of the sec_cm testplan for the V2S signoff.

**Comment by rswarbrick** [comment]
Oops, I forgot to close this! Yep, all done, I think.
